tools/nut-scanner/scan_snmp.c

```diff
@@ -855,6 +867,17 @@ nutscan_device_t * nutscan_scan_snmp(const char * start_ip, const char * stop_ip
 //		free(ip_str); // Do not free() here - seems to cause a double-free instead
 		ip_str = nutscan_ip_iter_inc(&ip);
 //		free(tmp_sec);
+            } else {
+#ifdef HAVE_PTHREAD
```

Comment:
Did you compile without pthread support? I think the empty clause `... } else { }` can be a problem - maybe the `}else{` line above should be moved into the `ifdef` too.